docs: improve model.rst (#6962)
codeigniter4/CodeIgniter4: user_guide_src/source/models/model.rst

.. note:: The ``$primaryKey`` field should never be an allowed field.

Dates
-----

$useTimestamps
^^^^^^^^^^^^^^

This boolean value determines whether the current date is automatically added to all inserts
and updates. If true, will set the current time in the format specified by ``$dateFormat``. This
requires that the table have columns named **created_at**, **updated_at** and **deleted_at** in the appropriate
data type.

$dateFormat
^^^^^^^^^^^

This value works with ``$useTimestamps`` and ``$useSoftDeletes`` to ensure that the correct type of
date value gets inserted into the database. By default, this creates DATETIME values, but

> valid options are: ``'datetime'``, ``'date'``, or ``'int'`` (a PHP timestamp). Using **useSoftDeletes** or
> useTimestamps with an invalid or missing dateFormat will cause an exception.

Reviewer: These concepts should either be all bold or none. 
```suggestion
valid options are: ``'datetime'``, ``'date'``, or ``'int'`` (a PHP timestamp). Using **useSoftDeletes** or
**useTimestamps** with an invalid or missing **dateFormat** will cause an exception.
```

$createdField
^^^^^^^^^^^^^

Specifies which database field to use for data record create timestamp.
Leave it empty to avoid updating it (even if ``$useTimestamps`` is enabled).

$updatedField
^^^^^^^^^^^^^

Specifies which database field should use for keep data record update timestamp.
Leave it empty to avoid update it (even ``$useTimestamps`` is enabled).

$deletedField
^^^^^^^^^^^^^

> Specifies which database field should use for soft deletions. See :ref:`model-use-soft-deletes`.

Reviewer: ```suggestion
Specifies which database field to use for soft deletions. See :ref:`model-use-soft-deletes`.
```

    :doc:`Query Builder <../database/query_builder>`.
    However, since Query Builder is not a Model, you cannot call methods of the Model.

Getting Query Builder for Another Table
=======================================

If you need access to another table, you can get another instance of the Query Builder.
Pass the table name in as a parameter, but be aware that this will **not** return
a shared instance:

.. literalinclude:: model/044.php

Mixing Methods of Query Builder and Model
=========================================

You can also use Query Builder methods and the Model's CRUD methods in the same chained call, allowing for
very elegant use:

.. literalinclude:: model/045.php

In this case, it operates on the shared instance of the Query Builder held by the model.

.. important:: The Model does not provide a perfect interface to the Query Builder.
    The Model and the Query Builder are separate classes with different purposes.
    They should not be expected to return the same data.

If the Query Builder returns a result, it is returned as is.
In that case, the result may be different from the one returned by the model's method

> and may not be what was expected. The model's event is not be triggered.

Reviewer: ```suggestion
and may not be what was expected. The model's events are not triggered.
```
